arxiv/metadata/metacheck.py

```diff
@@ -0,0 +1,687 @@
+
+import re
+
+from typing import Dict
+
+import gcld3
+
+from arxiv.authors import parse_author_affil
+
+from arxiv.metadata import FieldName
+from arxiv.metadata import Disposition
+from arxiv.metadata.all_caps_words import KNOWN_WORDS_IN_ALL_CAPS
+
+############################################################
+TITLE = FieldName.TITLE
+AUTHORS = FieldName.AUTHORS
+ABSTRACT = FieldName.ABSTRACT
+SOURCE_TYPE = FieldName.SOURCE_TYPE
+COMMENTS = FieldName.COMMENTS
+REPORT_NUM = FieldName.REPORT_NUM
+JOURNAL_REF = FieldName.JOURNAL_REF
+DOI = FieldName.DOI
+
+OK = Disposition.OK
+WARN = Disposition.WARN
+HOLD = Disposition.HOLD
+
+############################################################
+# Some constants
+
+MIN_TITLE_LEN = 5
+MIN_AUTHORS_LEN = 5
+MIN_ABSTRACT_LEN = 5
+MIN_SOURCE_TYPE_LEN = 0         # Not checked

```

Comment:
Source type is usually considered to be part of the author entered metadata.

Reply:
We're planning on moving in that direction, but source type is a fixed set of values in the db, iirc.